src/Workspaces/Core/Portable/CodeStyle/CodeStyleHelpers.cs

```diff
@@ -98,6 +98,7 @@ public static bool TryParseNotification(string value, out NotificationOption not
                     notification = NotificationOption.None;
                     return true;
 
+                case EditorConfigSeverityStrings.Refactoring:
```

Comment:
ok.  so this means in editor-config, people can do true:refactoring, yes?

Reply:
➡️ Yes, this is correct. It's the same as `true:silent` and is added simply to reduce confusion about terminology differences between Tools&rarr;Options and .editorconfig.

Reply:
I'm a little confused by this terminology. Refactorings are often available even if they go *against* the user preference (for example implicit/explicit type). Therefore I would expect "refactoring" to be equivalent to "none" as opposed to "silent".

Reply:
we find some customers want none as none. no squiggle, no suggestion, no error or no refactoring. they want to completely turn off these offering for whatever reasons they have. (some said having tool icon distract them too much for things they never use or care)

they can turn completely off by opening rule set editor and set it as none. they now got confused since they have 3 different things (ruleset, editorconfig, tools->options) for the same cause (issues) that have slightly different behavior.

this will make these 3 to have same behavior and give those cusomters a way to completely off these features they never want.

Reply:
I can understand that a refactoring icon may distract some people but isn't that an argument for adding an option to hide it as opposed to going through every single refactoring and setting it to none?

Reply:
My point is:
1. We get the refactoring even without setting this to "refactoring only". For example even if my preference is var, I always get a refactoring to use explicit type. Or am I mistaken? Does "none" actually disable the refactoring?
2. As I understand it, "silent" is more than a refactoring, for example it will participate in "fix all", is that true? Having "refactoring" as a synonym for "silent" is a little confusing then, because it's much more than that.

Reply:
What's the reason for adding "refactoring only"? I would just expect it to be changed to "Silent" in tools->options to match editorconfig. I find "refactoring" or "refactoring only" a little confusing (but maybe I don't understand the change).

Reply:
@Neme12 The behavior of "Silent" was not obvious to users. The behavior was more closely related to what users first thought when they heard "Refactoring". There are still some nuances and inconsistencies (e.g. there are cases where refactorings are still offered when `none` is used), but in general more users have a better understanding of the behavior with fewer words when the terms in this pull request are used.

Reply:
@kuhlenh Has a table showing the behavior of `none` and `silent`/`refactoring` which makes it quite clear where everything is expected to work. 😄

Reply:
> Does "none" actually disable the refactoring?

yes, it disables that particular one.

>, for example, it will participate in "fix all"

I believe you are making this distinction since you actually know how code fix/refactoring provider works. from feedbacks you get, most of the users don't/can't distinguish those except icon. they don't seem to even know/care what code fix provider/refactoring provider is.

hidden diagnostic's code fix shows up with refactoring icon even though it is not provided by refactoring provider. so they are the same thing for most users. so we are reflecting it to names.

two are slightly different in behavior and capability they provide. but so far, it looks like users don't care on those much. if we get more feedback/telemetry on those, we will improve on those in future.